### ISSUE TITLE
terraform init reconfigures backend every time

### ISSUE THREAD

**Issue by jeffmccollum** [hashicorp/terraform#21567]
### Terraform Version
```
...
Terraform v0.12.0
+ provider.aws v2.12.0
```

### Terraform Configuration Files

```hcl
...
terraform {
  required_version = ">= 0.12"

  backend "s3" {
    region         = "us-east-1"
    bucket         = "statebucket"
    key            = "vpc/terraform.tfstate"
    dynamodb_table = "locking-table"
  }
}
```

### Debug Output
```
2019/06/03 10:58:43 [INFO] Terraform version: 0.12.0
2019/06/03 10:58:43 [INFO] Go runtime version: go1.12.4
2019/06/03 10:58:43 [INFO] CLI args: []string{"/usr/local/bin/terraform12", "init"}
2019/06/03 10:58:43 [DEBUG] Attempting to open CLI config file: /home/jm/.terraformrc
2019/06/03 10:58:43 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019/06/03 10:58:43 [INFO] CLI command args: []string{"init"}
Initializing modules...
2019/06/03 10:58:43 [TRACE] ModuleInstaller: installing child modules for . into .terraform/modules
2019/06/03 10:58:43 [DEBUG] Module installer: begin vpc
2019/06/03 10:58:43 [TRACE] ModuleInstaller: Module installer: vpc <nil> already installed in .terraform/modules/vpc
2019/06/03 10:58:43 [TRACE] modsdir: writing modules manifest to .terraform/modules/modules.json

Initializing the backend...
2019/06/03 10:58:43 [TRACE] Meta.Backend: built configuration for "s3" backend with hash value 588198649
2019/06/03 10:58:43 [TRACE] Preserving existing state lineage "9e7a5b09-7be4-7d37-8b02-9e32a7f5caad"
2019/06/03 10:58:43 [TRACE] Preserving existing state lineage "9e7a5b09-7be4-7d37-8b02-9e32a7f5caad"
Backend configuration changed!
2019/06/03 10:58:43 [TRACE] Meta.Backend: working directory was previously initialized for "s3" backend

2019/06/03 10:58:43 [TRACE] backendConfigNeedsMigration: configuration values have changed, so migration is required
2019/06/03 10:58:43 [TRACE] Meta.Backend: backend configuration has changed (from type "s3" to type "s3")
2019/06/03 10:58:43 [WARN] backend config has changed since last init
Terraform has detected that the configuration specified for the backend
2019/06/03 10:58:43 [INFO] Setting AWS metadata API timeout to 100ms
has changed. Terraform will now check for existing state in the backends.


2019/06/03 10:58:44 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2019/06/03 10:58:44 [INFO] AWS Auth provider used: "SharedCredentialsProvider"
2019/06/03 10:58:44 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2019/06/03 10:58:44 [DEBUG] [aws-sdk-go] DEBUG: Request sts/GetCallerIdentity Details:
```
### Expected Behavior
Terraform init shouldn't detect any backend config changes

### Actual Behavior
```
Initializing modules...

Initializing the backend...
Backend configuration changed!

Terraform has detected that the configuration specified for the backend
has changed. Terraform will now check for existing state in the backends.


Do you want to migrate all workspaces to "s3"?
  Both the existing "s3" backend and the newly configured "s3" backend
  support workspaces. When migrating between backends, Terraform will copy
  all workspaces (with the same names). THIS WILL OVERWRITE any conflicting
  states in the destination.

  Terraform initialization doesn't currently migrate only select workspaces.
  If you want to migrate a select number of workspaces, you must manually
  pull and push those states.

  If you answer "yes", Terraform will migrate all states. If you answer
  "no", Terraform will abort.

  Enter a value:
```
### Steps to Reproduce

`terraform init`

### Additional Context
Running Terraform locally, and sourcing a module in Github

### References
This seems very similar to issue - #13149 however this started occurring with version 0.12. With version 0.11.14 this issue did not occur.

**Comment by jbardin** [comment]
Hi @jeffmccollum,

Thanks for filing the issue. This was fixed by #21439, and will be in the next release.

**Comment by jeffmccollum** [comment]
@jbardin Sorry for opening this issue, I didn't see that in my issue search. Glad to hear it is fixed, thanks!
